File: problem/2/python/solve_2.py
#!/usr/bin/env python
# http://projecteuler.net/problem=2

import sys

def even_fibonacci(n):
        a, b = 1, 2
        results = []
#        for i in range(0, n):
        while True:
                if a > n: break
                if a % 2 == 0:
                        results.append(a)
                a, b = b, a + b
        return results

print sum(even_fibonacci(int(sys.argv[1])))

File: problem/2/perl/solve_2.pl
#!/usr/bin/env perl6
# http://projecteuler.net/problem=2

use v6;

my ($a, $b) = (1, 2);
my $temp;
my @results := gather loop (;;) {
        if $a > @*ARGS[0] { last; }
        if $a % 2 == 0 { take $a; }
        $temp = $b;
        $b = $a + $b;
        $a = $temp;
}

say [+] @results;
